pkg/simulate/pushsync/pushsync.go

```diff
@@ -0,0 +1,305 @@
+	defer func(cancelID int) {
+		_ = proxy.Cancel(cancelID)
+	}(cancelID)
+
+	if err != nil {
+		return fmt.Errorf("proxy execute %w", err)
+	}
+	time.Sleep(time.Second * 5)
+
+	// set up clients
+	clients, err := cluster.NodesClients(ctx)
+	if err != nil {
+		return fmt.Errorf("node clients: %w", err)
+	}
+
+	if len(clients) <= 2 {
+		return errors.New("not enough nodes")
+	}
+
+	names := shuffle(rnd, cluster.NodeNames(), o.Seed)
+
+	//
+
+	buckets, leftovers := toBuckets(names, o.StartPercentage, o.EndPercentage, o.StepPercentage)
+
+	malfunctionEth := 0
+
+	for _, bucket := range buckets {
+
+		// STEP: freeze nodes in bucket
+		for _, n := range bucket {
+			node := clients[n]
+
+			nodeAddr, err := node.Addresses(ctx)
+			if err != nil {
+				return err
+			}
+
+			if len(nodeAddr.Underlay) == 0 {
+				return errors.New("not enough underlay addresses")
+			}
+
+			nodeIP := getIPFromUnderlays(nodeAddr.Underlay)
+			if nodeIP == "" {
+				continue
+			}
+
+			fmt.Printf("freezing block number for node %s ip %s\n", n, nodeIP)
+
+			cancelID, err := proxy.Execute(proxyClient.BlockNumberFreeze, nodeIP)
+			if err != nil {
+				return err
+			}
+
+			defer func(cancelID int) {
+				_ = proxy.Cancel(cancelID)
+			}(cancelID)
+		}
+
+		malfunctionEth += len(bucket)
+
+		// Upload nodes are not in the list of malfunctioning nodes
+		// so upload errors should not be expected
+
+		chunks := chunkBatch(rnd, int(o.ChunkCount))
+
+		index := rnd.Intn(len(leftovers))
+		uploadName := leftovers[index]
+		uploadNode := clients[uploadName]
+		fmt.Printf("using node %s as uploader\n", uploadName)
+
+		uploaded, err := uploadChunks(ctx, rnd, o, uploadNode, chunks)
+		if err != nil {
+			return fmt.Errorf("upload: %w", err)
+		}
+
+		metricStr := fmt.Sprintf("%d out of %d malfunctioning backends", malfunctionEth, len(names))
+		metrics.uploadedCounter.WithLabelValues(metricStr).Add(float64(uploaded))
+
+		var downloaded int
+		for i := 0; i < int(o.DownloadNodeAttemps); i++ {
+			downloadName := randomCmp(rnd, uploadName, names)
+			downloadNode := clients[downloadName]
+
+			fmt.Printf("using node %s as downloader\n", downloadName)
+
+			downloaded = downloadChunks(ctx, o, uploaded, downloadNode, chunks)
+
+			fmt.Printf("%s\n", metricStr)
+			fmt.Printf("uploaded to %s %d chunks\n", uploadName, uploaded)
+			fmt.Printf("downloaded from %s %d chunks\n", downloadName, downloaded)
+
+			metrics.downloadRetry.WithLabelValues(metricStr).Inc()
+
+			if downloaded == uploaded {
+				break
+			}
+		}
+
+		metrics.downloadedCounter.WithLabelValues(metricStr).Add(float64(downloaded))
+	}
+
+	return nil
+}
+
+func uploadChunks(ctx context.Context, rnd *rand.Rand, o Options, client *bee.Client, chunks []swarm.Chunk) (int, error) {
+
+	batchID, err := client.CreatePostageBatch(ctx, o.PostageAmount, o.PostageDepth, o.GasPrice, o.PostageLabel, false)
+	if err != nil {
+		return 0, fmt.Errorf("batch create %w", err)
+	}
+	fmt.Printf("batch id %s\n", batchID)
```

Comment:
not sure if this is needed. if debugging information is needed, the `verbose` option in `CreatePostageBatch` can be toggled